packages/flutter_tools/lib/src/run_cold.dart

```diff
@@ -195,6 +205,19 @@ class ColdRunner extends ResidentRunner {
           'An Observatory debugger and profiler on $dname is available at: '
           '${device.vmService.httpAddress}',
         );
+
+        final DevToolsServerAddress devToolsServerAddress = activeDevToolsServer();
+        if (devToolsServerAddress != null) {
+          final Uri uri = devToolsServerAddress.uri?.replace(
+            queryParameters: <String, dynamic>{'uri': '${device.vmService.httpAddress}'},
+          );
+          if (uri != null) {
```

Comment:
In the event there are multiple devices, this is going to get printed out `N` times. Hoist this out of the loop and print it at the end.

Reply:
if there are multiple devices, though, each with a different vmService uri, the uri will be different for each device

Reply:
Ahh, sorry - looking at this again I see that it is specific to each device.